app/shared/src/GeolocationContext.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+
+import * as React from 'react';
+
+const defaultState = {
+  lat: null,
+  lng: null,
+  canGetUserLocation: true,
+};
+
+const { Consumer, Provider: ContextProvider } = React.createContext({
+  ...defaultState,
+  actions: {
+    updateGeolocation: () => {},
+  },
+});
+
+type Props = {|
+  children: React.Node,
+|};
+
+type State = {|
+  lat: number | null,
+  lng: number | null,
+  canGetUserLocation: boolean,
+  actions: {|
+    updateGeolocation: () => void,
+  |},
+|};
+
+class Provider extends React.Component<Props, State> {
+  state = {
+    ...defaultState,
+    actions: {
+      updateGeolocation: this.updateGeolocation,
+    },
+  };
+  componentDidMount = () => {
+    this.updateGeolocation();
+  };
+
+  updateGeolocation = () => {
+    navigator.geolocation.getCurrentPosition(
```

Comment:
@tbergq Where is this `navigator` from?

Reply:
https://facebook.github.io/react-native/docs/geolocation.html

Reply:
> this API is available through the navigator.geolocation global - you do not need to import it.

Reply:
Should we set `geo_options` (timeout or maximumAge - dunno what's the difference)? When I change the location it will never return the new one and I think it's because of `maximumAge = INFINITY`...

Reply:
timeout is the the timeout for getting gps locations I guess. maxiumumAge sounds like, how long it will allow you to use lat, lng. Infinity sounds ok to me. I can set timeout.

Reply:
So you'll open the app in Prague and you'll have Prague there forever? (try to open the app, change the search location, go back to the homepage and go back to the search...)

Reply:
If you close the app, then open the app, you would get new gps coordinates.
You are always free to search for a location.

Since we have the app state change, we could always refresh gps coordinates there. Sound ok?